src/includes/load-service.h

```diff
@@ -957,7 +978,9 @@ class service_settings_wrapper
 
         // If socket_gid hasn't been explicitly set, but the socket_uid was specified as a name (and
         // we therefore recovered the primary group), use the primary group of the specified user.
+        // Same for logfile_uid/logfile_gid
         if (socket_gid == (gid_t)-1) socket_gid = socket_uid_gid;
+        if (logfile_gid == (gid_t)getgid()) logfile_gid = logfile_uid_gid;
```

Comment:
I think most issues are resolved but one remaining at least: If `logfile-uid` has been set, use primary group of that user as `logfile-gid` otherwise use value of `logfile-gid` or dinit process gid. In current implementation, logfile-gid completely ignored if `logfile-uid` has been set.  Hmm looks weird to fix.

Reply:
Why not use the same logic as for socket? -1 can be the special value to mean "not (yet) set". You can initialise `logfile_uid` and `logfile_gid` to -1, set `logfile_gid` to `logfile_uid_gid` if it's still -1 here, and set it to `getgid()` if it's still -1 after that (as well as setting `logfile_uid` to `getuid()` if it is -1).

In general, please try to use similar logic to other parts of the code when it makes sense. We shouldn't do one thing one way and a slightly different thing a completely different way, if they can be done the same way, unless there is a good reason.